autopilot/combinedattach.go

```diff
@@ -126,3 +127,23 @@ func (c *WeightedCombAttachment) NodeScores(g ChannelGraph, chans []Channel,
 
 	return scores, nil
 }
+
+// SetNodeScores is used to set the internal map from NodeIDs to scores if a
+// score settable sub-heuristic is active.
+//
+// NOTE: This is a part of the ScoreSettable interface.
+func (c *WeightedCombAttachment) SetNodeScores(
+	newScores map[NodeID]float64) error {
+
+	// Check if the external score attachement is active.
+	for _, h := range c.heuristics {
+		s, ok := h.AttachmentHeuristic.(ScoreSettable)
+		if !ok {
+			continue
+		}
+
+		return s.SetNodeScores(newScores)
```

Comment:
This only sets the first score found, though I’d expect this method forward the call to all relevant sub-heuristics. Note that the existing behavior is non-deterministic due to map range order if multiple mutable heuristics are present

Reply:
This was a simplification I made since we are only expecting at most one such heuristic at this point. I'm warming up to making it possible to target sub-heuristics with a name parameter....

Reply:
Fixed.